data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IsmPolicyManagement.java

```diff
@@ -93,10 +94,19 @@ public IsmPolicyManagement(final OpenSearchClient openSearchClient,
     }
 
     @Override
-    public Optional<String> checkAndCreatePolicy() throws IOException {
+    public Optional<String> checkAndCreatePolicy(final String indexAlias) throws IOException {
         final String policyManagementEndpoint = POLICY_MANAGEMENT_ENDPOINT + policyName;
 
         String policyJsonString = retrievePolicyJsonString(policyFile);
+	if(!indexAlias.isEmpty()) {
```

Comment:
Please add unit tests to cover this scenario.

Reply:
Upon checking the code (I am not sure per good in Java) there is a beautiful testcase testInstantiateSinkCustomIndex_WithIsmPolicy which covers this already. Since we add the index-patterns dynamically based on indexAlias (with or without variables) we need to change the test template json.file content to the following.

./data-prepper-plugins/opensearch/src/test/resources/test-custom-index-policy-file.json

`"ism_template": {
      "priority": 100,
      "index_patterns": ["dummy-pattern-*"]
    }
`
And as part the test case we still expect the policy name to be indexAlias + "-policy". So, we are good.